elide-core/src/main/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialect.java

```diff
@@ -0,0 +1,315 @@
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.filter.dialect;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.Operator;
+import com.yahoo.elide.core.filter.Predicate;
+import com.yahoo.elide.core.filter.expression.AndFilterExpression;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.core.filter.expression.NotFilterExpression;
+import com.yahoo.elide.core.filter.expression.OrFilterExpression;
+import com.yahoo.elide.utils.coerce.CoerceUtil;
+import cz.jirutka.rsql.parser.RSQLParser;
+import cz.jirutka.rsql.parser.RSQLParserException;
+import cz.jirutka.rsql.parser.ast.AndNode;
+import cz.jirutka.rsql.parser.ast.ComparisonNode;
+import cz.jirutka.rsql.parser.ast.ComparisonOperator;
+import cz.jirutka.rsql.parser.ast.Node;
+import cz.jirutka.rsql.parser.ast.OrNode;
+import cz.jirutka.rsql.parser.ast.RSQLOperators;
+import cz.jirutka.rsql.parser.ast.RSQLVisitor;
+import lombok.Getter;
+
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * FilterDialect which implements support for RSQL filter dialect.
+ */
+public class RSQLFilterDialect implements SubqueryFilterDialect, JoinFilterDialect {
+
+    private final Pattern typedFilterPattern;
+    private final RSQLParser parser;
+    private final EntityDictionary dictionary;
+    private final Map<ComparisonOperator, Operator> operatorMap;
+
+    public RSQLFilterDialect(EntityDictionary dictionary, Set<ComparisonOperator> rsqlOperators) {
+        // Match "filter[<type>]"
+        typedFilterPattern = Pattern.compile("filter\\[([^\\]]+)\\]");
+
+        parser = new RSQLParser(rsqlOperators);
+        this. dictionary = dictionary;
+
+        /* Subuset of operators that map directly to Elide operators */
+        operatorMap = new HashMap<>();
+        operatorMap.put(RSQLOperators.IN, Operator.IN);
+        operatorMap.put(RSQLOperators.NOT_IN, Operator.NOT);
+        operatorMap.put(RSQLOperators.LESS_THAN, Operator.LT);
+        operatorMap.put(RSQLOperators.GREATER_THAN, Operator.GT);
+        operatorMap.put(RSQLOperators.GREATER_THAN_OR_EQUAL, Operator.GE);
+        operatorMap.put(RSQLOperators.LESS_THAN_OR_EQUAL, Operator.LE);
+    }
+
+    public RSQLFilterDialect(EntityDictionary dictionary) {
+        this(dictionary, RSQLOperators.defaultOperators());
+    }
+
+
+    @Override
+    public FilterExpression parseGlobalExpression(String path, MultivaluedMap<String, String> queryParams)
+            throws ParseException {
+        if (queryParams.size() != 1) {
+            throw new ParseException("There can only be a single filter query parameter");
```

Comment:
is `queryParams` just the subset of params that start with `filter` or do we mean that you can't use RSql filters with sparse fieldsets, includes, and other features?